utils/src/test/java/com/cloud/utils/UriUtilsTest.java

```diff
@@ -273,4 +273,16 @@ public void testIsUrlForCompressedFile() {
         Assert.assertTrue(UriUtils.isUrlForCompressedFile("https://abc.com/xyz.gz"));
         Assert.assertFalse(UriUtils.isUrlForCompressedFile("http://abc.com/xyz.qcow2"));
     }
+
+    @Test
+    public void validateUrl() {
+        Pair<String, Integer> url1 = UriUtils.validateUrl("https://www.cloudstack.org");
+        Assert.assertEquals(url1.first(), "www.cloudstack.org");
+
+        Pair<String, Integer> url2 = UriUtils.validateUrl("https://www.apache.org");
+        Assert.assertEquals(url2.first(), "www.apache.org");
+
+        Pair<String, Integer> url3 = UriUtils.validateUrl("https://ipv6.google.com");
+        Assert.assertEquals(url3.first(), "ipv6.google.com");
```

Comment:
This testcase is failing during BO build. Do we need to add any guard for ipv6 support? cc @wido 

```
13:18:52 [INFO] Running com.cloud.utils.UriUtilsTest
13:18:52 [ERROR] Tests run: 15, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.139 s <<< FAILURE! - in com.cloud.utils.UriUtilsTest
13:18:52 [ERROR] validateUrl(com.cloud.utils.UriUtilsTest)  Time elapsed: 0.113 s  <<< ERROR!
13:18:52 java.lang.IllegalArgumentException: Unable to resolve ipv6.google.com
13:18:52 	at com.cloud.utils.UriUtilsTest.validateUrl(UriUtilsTest.java:285)
```

Reply:
We need to make sure that this code can work with IPv6 as well. But the problem is that the host where the build is running on doesn't support IPv6 and thus fails to run the test properly. This also explains why it doesn't fail on my local machines as they do have IPv6.

Reply:
@wido could you write the test such that it can skip if host does not have ipv6 enabled?

Reply:
> @wido could you write the test such that it can skip if host does not have ipv6 enabled? But still work in env like Github Actions where ipv6 is available

That's something you can't detect. I think we would have to catch the IllegalArgumentException in this case

Reply:
Maybe by checking if any of the network interfaces on the system have an IPv6 address?

Reply:
> Maybe by checking if any of the network interfaces on the system have an IPv6 address?

That wouldn't give you a real idea if IPv6 is available yes or now. In the end you can't know, you just have to try and see what happens.

Reply:
Is it possible to enable ipv6 on the hosts where this test is run?